server/plugin/bot_test.go

```diff
@@ -13,34 +13,46 @@ import (
 func TestDM(t *testing.T) {
 	p := Plugin{}
 	for _, testCase := range []struct {
-		description    string
-		mockChannelErr *model.AppError
-		mockPostErr    *model.AppError
+		description   string
+		setupAPI      func(*plugintest.API)
+		expectedError string
 	}{
 		{
 			description: "DM: message is successfully posted",
+			setupAPI: func(a *plugintest.API) {
+				a.On("LogError", testutils.GetMockArgumentsWithType("string", 5)...).Return(nil)
+				a.On("LogError", testutils.GetMockArgumentsWithType("string", 3)...).Return(nil)
+				a.On("GetDirectChannel", mock.Anything, mock.Anything).Return(testutils.GetChannel(model.CHANNEL_PRIVATE), nil)
+				a.On("CreatePost", mock.Anything).Return(testutils.GetPost(), nil)
+			},
 		},
 		{
-			description:    "DM: channel is not found",
-			mockChannelErr: &model.AppError{},
+			description: "DM: channel is not found",
+			setupAPI: func(a *plugintest.API) {
+				a.On("LogError", testutils.GetMockArgumentsWithType("string", 5)...).Return("LogError error")
```

Comment:
```suggestion
				a.On("LogError", testutils.GetMockArgumentsWithType("string", 5)...).Return()
```
LogError has no return type.